### PR TITLE
Proxy to Public Annotations API

### PR DESCRIPTION
**Pull request by peteclark-ft**

### CODE REVIEW

File: annotations/annotations_api.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	apiResp, err := api.httpClient.Do(apiReq)
+	if err != nil {
+		return fmt.Errorf("gtg call error: %v", err.Error())
+	}
+	defer apiResp.Body.Close()
+
+	if apiResp.StatusCode != http.StatusOK {
+		errMsgBody, err := ioutil.ReadAll(apiResp.Body)
+		if err != nil {
+			return errors.New("gtg returned a non-200 HTTP status")
```
Comment (khatton-ft) [inline annotations/annotations_api.go]: But you could still include apiResp.StatusCode in the error message.

File: health/healthcheck.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package health
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Financial-Times/draft-annotations-api/annotations"
+	health "github.com/Financial-Times/go-fthealth/v1_1"
```
Comment (khatton-ft) [inline health/healthcheck.go]: I'm surprised this works, and it's downright confusing. Maybe alias this package to `fthealth` for clarity?

File: README.md [hunk 3]
```diff
@@ -1,109 +1,100 @@
...
+[![Circle CI](https://circleci.com/gh/Financial-Times/draft-annotations-api/tree/master.png?style=shield)](https://circleci.com/gh/Financial-Times/draft-annotations-api/tree/master)[![Go Report Card](https://goreportcard.com/badge/github.com/Financial-Times/draft-annotations-api)](https://goreportcard.com/report/github.com/Financial-Times/draft-annotations-api) [![Coverage Status](https://coveralls.io/repos/github/Financial-Times/draft-annotations-api/badge.svg)](https://coveralls.io/github/Financial-Times/draft-annotations-api)
 
 ## Introduction
 
-_What is this service and what is it for? What other services does it depend on_
-
-UPP Golang Microservice Template short description - please amend
+Draft Annotations API is a microservice that provides access to draft annotations for content stored in PAC. Currently, the service is a simple proxy to UPP Public Annoations API.
```
Comment (khatton-ft) [inline README.md]: s/Annoations/Annotations

File: README.md [hunk 4]
```diff
@@ -1,109 +1,100 @@
...
 Download the source code, dependencies and test dependencies:
 
-        go get -u github.com/kardianos/govendor
-        go get -u github.com/Financial-Times/content-editorial-read
-        cd $GOPATH/src/github.com/Financial-Times/content-editorial-read
-        govendor sync
-        go build .
+```
+go get -u github.com/kardianos/govendor
+go get -u github.com/Financial-Times/draft-annotations-api
```
Comment (khatton-ft) [inline README.md]: Actually this is probably _not_ what you want to do. At this point do a standard `git clone`, then use `govendor sync` as below to pull in the dependencies.

File: README.md [hunk 5]
```diff
@@ -1,109 +1,100 @@
...
-_How can I run it_
 
 1. Run the tests and install the binary:
 
-        govendor sync
-        govendor test -v -race
-        go install
+```
+govendor sync
+govendor test -v -race
```
Comment (khatton-ft) [inline README.md]: Does this imply `./...`?

File: README.md [hunk 6]
```diff
@@ -1,109 +1,100 @@
...
+```
+http GET http://localhost:8080/draft/content/b7b871f6-8a89-11e4-8e24-00144feabdc0/annotations
+```
 
+Currently, this endpoint is a proxy to the annotations available in UPP, so it returns a payload consistent to the UPP Public Annotations API.
 
 ## Utility endpoints
-_Endpoints that are there for support or testing, e.g read endpoints on the writers_
+
+The standard endpoint for Go profiling is available under the path `/debug/pprof`. Details regarding Go profiling are available [here](https://golang.org/pkg/net/http/pprof/)
```
Comment (khatton-ft) [inline README.md]: We took it out of draft-content-api, so ...

File: main.go [hunk 7]
```diff
@@ -38,18 +42,40 @@ func main() {
...
 
 	app.Action = func() {
 		log.Infof("System code: %s, App Name: %s, Port: %s", *appSystemCode, *appName, *port)
 
+		annotationsAPI := annotations.NewAnnotationsAPI(*annotationsEndpoint, *uppAPIKey)
+		annotationsHandler := annotations.NewHandler(annotationsAPI)
+		healthService := health.NewHealthService(*appSystemCode, *appName, appDescription, annotationsAPI)
 		go func() {
-			serveAdminEndpoints(*appSystemCode, *appName, *port)
+			serveEndpoints(*port, apiYml, annotationsHandler, healthService)
```
Comment (peteclark-ft) [inline main.go]: This is taken from @izzyblues - but I'm assuming that waiting for certain os signals leads to a more graceful shutdown?

File: README.md (hunk 4 above)
Comment (peteclark-ft) [inline README.md]: That's not what we do in our circle or docker builds?

Comment (khatton-ft) [inline README.md]: Docker and Circle are both running in the context of the checked out project anyway. (It's just the second `go get` I'm flagging, not the govendor one.)

File: annotations/annotations_api.go [hunk 8]
```diff
@@ -36,22 +35,22 @@ func (api *annotationsAPI) Get(ctx context.Context, contentUUID string) (*http.R
 	apiReqURI := fmt.Sprintf(api.endpointTemplate, contentUUID)
 	getAnnotationsLog := log.WithField("url", apiReqURI).WithField("uuid", contentUUID)
 
-	tID, err := tidutils.GetTransactionIDFromContext(ctx)
+	tid, err := tidutils.GetTransactionIDFromContext(ctx)
 	if err != nil {
-		getAnnotationsLog.WithField(tidutils.TransactionIDKey, "TID Not found: "+err.Error())
-	} else {
-		getAnnotationsLog = getAnnotationsLog.WithField(tidutils.TransactionIDKey, tID)
+		tid = "not found"
```
Comment (khatton-ft) [inline annotations/annotations_api.go]: I'd use a value without whitespace, just to discourage logrus from being creative with quotation marks.

**Pull request merged**